packages/headless/src/api/form-url-encoder.ts

```diff
@@ -0,0 +1,33 @@
+type FormEncodable = Record<string, string | number | boolean>;
+
+export function encodeAsFormUrl(obj: FormEncodable) {
+  const body = [];
+
+  for (const property in obj) {
+    const key = encodeURIComponent(property);
+    const value = encodeURIComponent(obj[property]);
+    body.push(`${key}=${value}`);
+  }
+
+  return body.join('&');
+}
```

Comment:
I added our own function to form-url encode after [seeing this comment](https://stackoverflow.com/questions/35325370/how-do-i-post-a-x-www-form-urlencoded-request-using-fetch).

While node and the browser offer a `URLSearchParams` utility class, other javascript run-times such as ReactNative do not.

Reply:
There's also https://developer.mozilla.org/en-US/docs/Web/API/FormData you could use.

But similar problems could (will?) happen, with discrepancy in Node vs Browser vs React Native vs Locker Service etc.

Such are the woes of JavaScript...

Reply:
Yep, I am not sure. I found some links:

1. https://stackoverflow.com/questions/32441963/how-to-use-formdata-in-react-native
2. https://aboutreact.com/file-uploading-in-react-native/

It seems like the `FormData` class is not available in react-native core, but can be added by installing additional dependencies.

My gut feeling though is it will serve us well to minimize use of global classes. This should help keep headless flexible and able to run in more places. Avoiding third party dependencies will also avoid bloat and needing to maintain the dependency when there are security vulnerabilities, breaking changes etc.